scripts/biomedical/diseasesAtJensenLab/scripts/format_disease_jensen_lab.py

```diff
@@ -0,0 +1,383 @@
+    df = format_doid_icd(df)  # Format DOID and ICD10 columns
+    df = format_disease_gene_cols(df, data_type)  # Format disease and gene columns
+    df = df.apply(lambda x: check_for_dcid(x), axis=1)  # Check DCID-related columns for illegal characters
+    return df
+
+def format_data_type_specific_info(df, data_type):
+    """Adds data type-specific information (e.g., association type)."""
+    if data_type == 'experiments':
+        df['associationType'] = 'dcs:AssociationSourceExperiment'
+        df['source-score'] = df['source-score'].str.split('=')  # Split source score string
+        df['source-score'] = np.where(df['source-score'] == df['source-score'], df['source-score'].str, np.nan) # Extract the score value
+        df.update('"' + df[['Disease_Name', 'score-db']].astype(str) + '"')  # Add quotes around specific columns
+    if data_type == 'knowledge':
+        df['associationType'] = 'dcs:AssociationSourceKnowledge'
+        df.update('"' + df[['Disease_Name', 'score-db']].astype(str) + '"')  # Add quotes
+    if data_type == 'textmining':
+        df['associationType'] = 'dcs:AssociationSourceTextMining'
+        df.update('"' + df[['Disease_Name', 'url']].astype(str) + '"')  # Add quotes
+    return df
+
+
+def clean_data(df, data_type):
+	"""
+    Cleans and separates a DataFrame into coding and non-coding RNA data.
+
+    Args:
+        df: The input DataFrame containing RNA data.
+        data_type: The type of data being processed (used by formatting functions).
+
+    Returns:
+        A tuple containing two DataFrames:
+            - df: DataFrame with coding RNA data (and some gene data from non-coding).
+            - df_non_coding: DataFrame with only non-coding RNA data.
+    """
+	df_non_coding = df  # Create a copy to hold non-coding RNA data
+	searchfor = ['ENSP00', 'LINC', 'linc'] # list of stings to identify non-coding RNAs
+	
+	# Filter df to keep only coding RNAs (containing "ENSP00" and not in searchfor)
+	df = df[df['Id'].str.contains("ENSP00")]
+	df = df[~df.Gene.str.contains('|'.join(searchfor))]
+
+	# Filter df_non_coding to keep only non-coding RNAs (not present in the filtered df)
+	df_non_coding = df_non_coding[~df_non_coding.isin(df)].dropna() ## df with only non coding RNAs
+	
+	# Further filter non-coding RNAs to remove unwanted entries (containing "chr" or "ENSP00")
+	df_non_coding = df_non_coding[~df_non_coding['Gene'].str.contains("chr")]
+	df_non_coding = df_non_coding[~df_non_coding['Gene'].str.contains("ENSP00")]
+	
+	# Format DCIDs (Data Center IDs) for both DataFrames
+	df = format_dcids(df, data_type)
+	df_non_coding = format_dcids(df_non_coding, data_type)
+	
+	# Format RNA types for the non-coding DataFrame
+	df_non_coding = format_RNA_type(df_non_coding)
+	
+	# Extract gene data from the non-coding DataFrame
+	df_gene = df_non_coding.loc[df_non_coding['RNA_type']=='Gene']
+	
+	# Remove gene entries from the non-coding DataFrame
```

Comment:
nit: can we do all the steps involving df_gene (lines 325-332) first and then go back to processing df_non_coding?